filebeat/beater/filebeat.go

```diff
@@ -146,20 +146,30 @@ func New(b *beat.Beat, rawConfig *common.Config) (beat.Beater, error) {
 	return fb, nil
 }
 
+// setupPipelineLoaderCallback sets the callback function for loading pipelines during setup.
 func (fb *Filebeat) setupPipelineLoaderCallback(b *beat.Beat) error {
-	if !fb.moduleRegistry.Empty() {
-		overwritePipelines := fb.config.OverwritePipelines
-		if b.InSetupCmd {
-			overwritePipelines = true
+	if b.Config.Output.Name() != "elasticsearch" {
+		logp.Warn(pipelinesWarning)
+		return nil
```

Comment:
Should this check be moved inside the callback in case the output can be dynamically configured at some moment?

Reply:
I guess this is not needed if this method is only used by the setup command as discussed in [this comment](https://github.com/elastic/beats/pull/12340/files#r290158124).

Reply:
I am not sure that during runnig `setup` there might be a point when ES suddenly becomes reachable.